src/python/pants/backend/jvm/tasks/coursier/coursier_subsystem.py

```diff
@@ -0,0 +1,83 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import hashlib
+import logging
+import os
+
+from pants.base.build_environment import get_buildroot
+from pants.base.workunit import WorkUnit, WorkUnitLabel
+from pants.java.distribution.distribution import DistributionLocator
+from pants.net.http.fetcher import Fetcher
+from pants.subsystem.subsystem import Subsystem
+from pants.util.dirutil import safe_concurrent_creation
+
+
+logger = logging.getLogger(__name__)
+
+
+class CoursierSubsystem(Subsystem):
+  """Common configuration items for coursier tasks.
+
+  :API: public
+  """
+  options_scope = 'coursier'
+
+  class Error(Exception):
+    """Indicates an error bootstrapping coursier."""
+
+  @classmethod
+  def register_options(cls, register):
+    super(CoursierSubsystem, cls).register_options(register)
+    register('--fetch-options', type=list, fingerprint=True,
+             help='Additional options to pass to coursier fetch. See `coursier fetch --help`')
+    register('--bootstrap-jar-url', fingerprint=True,
```

Comment:
I don't think the URL should be fingerprinted. Would recommend swapping the `version` string for an SHA256 fingerprint of the downloaded jar.

Alternatively, skip the fingerprinting for now and switch to fetching this with `binary_utils`, so that when we add support for fingerprinting binaries there, we get it here for free.